src/Components/Web/src/Forms/InputBase.cs

```diff
@@ -241,6 +242,12 @@ public override Task SetParametersAsync(ParameterView parameters)
             {
                 EditContext = CascadedEditContext;
                 EditContext.OnValidationStateChanged += _validationStateChangedHandler;
+                _shouldGenerateFieldNames = EditContext.ShouldUseFieldIdentifiers;
+            }
+            else
+            {
+                // Ideally we'd know if we were in an SSR context but we don't
+                _shouldGenerateFieldNames = !OperatingSystem.IsBrowser();
```

Comment:
We should do this always or have a way to configure it in webassembly (as it was possible with EditContext). Otherwise people will start on server, write CSS rules using name, etc. and then suddenly those things will start to fail in webassembly.

Reply:
> We should do this always or have a way to configure it in webassembly (as it was possible with EditContext)

It is still configurable in WebAssembly, as the EditContext approach still works. People who need the field names in WebAssembly for some reason can still use this to configure it on a per-area basis.

Reply:
The `!OperatingSystem.IsBrowser()` default is still in `EditContext` as a default, but there also needs to be a default in `InputBase` for the case where there is no `EditContext`.